src/base/snark0.ml

```diff
@@ -724,6 +724,10 @@ module Run = struct
            ~eval_constraints:false ~num_inputs:0 ~next_auxiliary:(ref 0)
            ~with_witness:false ~stack:[] ~is_running:false () )
 
+    let get_state () = !state
```

Comment:
I would much prefer not to spew the internals here. Can you use the callback pattern, like [here](https://github.com/o1-labs/snarky/pull/836), instead?